src/compiler/diagnosticMessages.json

```
@@ -7976,5 +7976,13 @@
     "'await using' statements cannot be used inside a class static block.": {
         "category": "Error",
         "code": 18054
+    },
+    "A string member initializer in a enum declaration can only use constant expressions when 'isolatedModules' is enabled.": {

```

Paraphrasing a suggestion from @DanielRosenwasser:

```suggestion
    "'{0}' has a string type, but must have syntactically recognizable string syntax when 'isolatedModules' is enabled.": {
```

We could totally do a quick fix to wrap it in a template literal but I think this error is going to be so niche that I’m not sure it’s worth it.

---

I like the suggested error message. But what do I substitute `{0}` with? The initializer can be a relatively complex expression. Is there a utility function to stringify that? Or should I use the enum member name or even the computed value?

---

I forgot to include that; Daniel’s suggestion was `'EnumName.MemberName'`

---

Sounds good. Do we want to handle member names that are not valid identifier? E.g.:

```ts
import {bar} from './bar';
enum Foo { ['not an identifier'] = bar }
// 'Foo.not an identifier' has a string type, ...
// 'Foo["not an identifier"]' has a string type, ...
```

---

I updated the error message and ignored the non-identifier case for now, because it seems like an extremely rare case. That means it would print `'Foo.not an identifier'` in the error message.